### ISSUE TITLE
Minecraft 1.11.2 support

### ISSUE THREAD

**Issue by MrPowerGamerBR** [PocketDreams/SequinLand#24]
https://github.com/PocketDreams/SequinLand/tree/minecraft-1112-support

This is very low priority, and it is "just for fun" and probably I will never finish it due to bugs and difficulties related to packets *remembers how chunks works* *closes Eclipse because chunks are difficult*

**Comment by MrPowerGamerBR** [comment]
![http://i.imgur.com/JjLenPm.png](http://i.imgur.com/JjLenPm.png)

**Comment by MrPowerGamerBR** [comment]
New branch (I decided to refactor the code) https://github.com/PocketDreams/SequinLand/tree/minecraft-1112-support-v2

**Comment by DaMatrix** [comment]
I can quite literally say that I am amazed. _quickly switches over to Google play to see if there's a portable IDE that supports maven_

But seriously, that's impressive. Should I get the Glowstone devs in here? They seem helpful enough

**Comment by MrPowerGamerBR** [comment]
@DaMatrix but why get them here? I don't have issues with packets (yet!)

The only issue is trying to parse the MCPE data chunk, because, while the block IDs parsing is working, block meta parsing is broken.

Take a look at this pixelart:
![http://i.imgur.com/7lFgVZM.png](http://i.imgur.com/7lFgVZM.png)

And this proves my theory that doing PC -> PE is easier than PE -> PC. 😉

**Comment by DaMatrix** [comment]
Are you sure it's the translator? It would seem strange if it only translated some blocks but not others.

**Comment by MrPowerGamerBR** [comment]
@DaMatrix it is the way metadata is processed (and how MCProtocolLib handles invalid meta)

Most of the meta are using high as fuck values, but MCProtocolLib fixes that for some reason. (but not for everything), if I set the metadata to 0, the blocks aren't invisible. (it is difficult to see in that screenshot because it is at night, but there is a lot of missing blocks in that pixelart)

**Comment by MrPowerGamerBR** [comment]
Also, tried implementing player movement....

![http://i.imgur.com/t81mS8O.png](http://i.imgur.com/t81mS8O.png)

*Close enough.*

![http://i.imgur.com/wkBiImB.png](http://i.imgur.com/wkBiImB.png) (the PC client can't view other players yet)

**Comment by MrPowerGamerBR** [comment]
![http://i.imgur.com/14UZDdR.gifv](http://i.imgur.com/14UZDdR.gifv)

**Comment by MrPowerGamerBR** [comment]
![http://i.imgur.com/9c2tYv9.gif](http://i.imgur.com/9c2tYv9.gif)

**Comment by MrPowerGamerBR** [comment]
The PC client is a bit wonky in the PE server because I didn't create a translator for movement/rotation packets.

**Comment by DaMatrix** [comment]
Well, it looks amazing. I'm gonna go ahead and mess around with this tomorrow, but I just want to congratulate you first for making so much progress so fast :D

**Comment by MrPowerGamerBR** [comment]
:wink:

![http://i.imgur.com/1yDyYy2.png](http://i.imgur.com/1yDyYy2.png)

**Comment by MrPowerGamerBR** [comment]
Block metadata is properly decoded now, (however the tall grass is still wrong, maybe it uses a different metadata or ID? need to double check later)

**Comment by MrPowerGamerBR** [comment]
Just for fun I will try adding MCPE support to Paper 1.11.2 (if it is easy and if I figure out how to compile it) and, if adding to Paper 1.11.2 is too difficult, I will try adding to Glowstone. (I was viewing Glowstone's source code and it seems easy to override the send/receive packet functions)

**Comment by DaMatrix** [comment]
I definitely like the idea of that better (full features). But if you do decide to do it, I would go with Glowstone (faster, despite some lack of features)

**Comment by DaMatrix** [comment]
I can't build the project, it would seem to me that you're using a custom build of MCProtocolLib?
![screenshot from 2017-02-23 17-24-20](https://cloud.githubusercontent.com/assets/11216106/23268083/f10a666a-f9ec-11e6-8a48-7299cce42e61.png)

`net.pocketdreams.mcprotocollib`

**Comment by MrPowerGamerBR** [comment]
@DaMatrix yeah, I needed to relocate MCProtocolLib's dependencies to not cause conflict with Nukkit.

You need to add that dependency in your maven local repo. (look how LevelDB is added and do the same thing)

**Comment by DaMatrix** [comment]
ah ok, thanks :D

**Comment by DaMatrix** [comment]
Well, it builds but i can only connect from PC. from PE i get nothing in the log and the client simply says "unable to connect to server".

**Comment by MrPowerGamerBR** [comment]
@DaMatrix is there's any errors in the console startup related to bind? Also, if you are using MCW10E, open the server first, then open MCW10E

**Comment by MrPowerGamerBR** [comment]
Glowstone-related work is here: https://github.com/PocketDreams/GlowstonePocket

Again, that is also a very very very WIP, you can login into the server and talk in that chat and that's it.
